src/pyprojroot/criterion.py

```diff
@@ -9,35 +9,50 @@
 import typing
 from os import PathLike as _PathLike
 
+
+_PathType = typing.Union[_PathLike, str]
+_CriterionType = typing.Union[
+    typing.Callable[[_PathType], bool],
+    typing.Callable[[_pathlib.Path], bool],
```

Comment:
I don't think you need to duplicate these for `_pathlib.Path`, since that's also "PathLike"